src/main/java/luke/commands/ReadCommand.java

```diff
@@ -0,0 +1,47 @@
+package luke.commands;
+
+import luke.data.TaskList;
+
+/**
+ * Framework for commands that reads the task list.
+ */
+public abstract class ReadCommand extends Command {
+
+    /**
+     * Takes in a task list and return the appropriate message based on the items in the task.
+     *
+     * @param taskList The task list to loop through.
+     * @param emptyMsg The message to print if the task list is empty.
+     * @param startMsg The message to prefix the task list with if the task list is not empty.
+     * @return The command result of the items in the task list.
+     */
+    public CommandResult readTaskList(TaskList taskList, String emptyMsg, String startMsg) {
+        return readTaskList(taskList, emptyMsg, startMsg, false);
+    }
+
+    /**
+     * Takes in a task list and return the appropriate message based on the items in the task.
+     *
+     * @param taskList   The task list to loop through.
```

Comment:
I like how most of your variable names and method names are self-explanatory. The naming is sufficient for reader to know what is going on in your code.